bravado/mapping/schema.py

```diff
@@ -0,0 +1,82 @@
+from bravado.exception import SwaggerError
+from bravado import swagger_type
+
+
+JSONSCHEMA_PRIMITIVE_VALIDATIONS = {
+    'string': ('minLength', 'maxLength', 'pattern', 'format', 'enum', 'default'),
+    'integer': ('multipleOf', 'minimum', 'maximum', 'exclusiveMaximum', 'exclusiveMinimum', 'enum', 'default', 'format'),
+    'number': ('multipleOf', 'minimum', 'maximum', 'exclusiveMaximum', 'exclusiveMinimum', 'enum', 'default', 'format'),
+    'boolean': ('enum', 'default'),
+    'null': (),
+}
+
+
+def to_primitive_schema(primitive_spec):
+    """Given the swagger spec for a primitive type, create the equivalent
+    jsonschema object for the primitive type.
+
+    :type primitive_spec: dict
+    :returns: jsonschema object
+    :rtype; dict
+    """
+    # identify the validation keys in the swager spec that are supported by
+    # jsonschema for the given primitive type
+    primitive_type = primitive_spec['type']
+    transferable_keys = ['name', 'type', 'description']
+    transferable_keys += JSONSCHEMA_PRIMITIVE_VALIDATIONS[primitive_type]
+    schema = {}
+    for key, value in primitive_spec.iteritems():
+        if key in transferable_keys:
+            schema[key] = value
+    return schema
```

Comment:
swagger supports more types than json, so this isn't going to cover many of the cases. On top of that I think this is more work (and a lot slower), than just checking the types directly.

We're going to want to be able to support user-defined formats as well, which won't work with this model, but should be a simple extension if we have our own `type` -> `validator` mapping already.